### PR TITLE
Implement Range#min, #max and #minmax

### PR DESCRIPTION
**Pull request by richardboehme**
This also fixes some issues with our spec library and a bug with implicit block passing for `super`.

[#555]

### CODE REVIEW

File: test/support/spec.rb [hunk 1]
```diff
@@ -763,11 +791,11 @@ def at_least(n)
     # FIXME: support endless ranges
```
Comment (richardboehme) [inline test/support/spec.rb:791]: Reminder: Remove this line

File: test/support/spec.rb [hunk 2]
```diff
@@ -763,11 +791,11 @@ def at_least(n)
...
     case n
     when :once
-      @count_restriction = 1..9_999_999
+      @count_restriction = 1..
     when :twice
-      @count_restriction = 2..9_999_999
+      @count_restriction = 2..
     else
-      @count_restriction = n..9_999_999
+      @count_restriction = n..
```
Comment (richardboehme) [inline test/support/spec.rb:798]: Those currently cause the CI to fail... I'm going to investigate the issue 😄

Comment (richardboehme) [reply test/support/spec.rb:798]: This seems to be related to NatalieParser. I opened natalie-lang/natalie_parser#4 with more information!

**Comment by seven1m** [comment]
I pushed some fixes to NatalieParser... let's see if that helps!

**Comment by richardboehme** [comment]
@seven1m I may need some help from you here... I tried debugging this one failing test when running with clang and I can't wrap my head around the issue. It seems to be GC related and I think it has something to do with ivars on the Float::CanCoerce object that gets used by the failing spec but I could not find the exact issue there. Do you have an idea?

**Comment by seven1m** [comment]
Well somehow I broke it worse. 😅 I'll dig in some more later today...

**Comment by seven1m** [comment]
I honestly don't know what I did. But at least I feel better that we have more guards in place to prevent capturing synthesized values in ivars. :man_shrugging:

**Comment by seven1m** [comment]
@richardboehme this is good to merge now yes?

**Comment by richardboehme** [comment]
> @richardboehme this is good to merge now yes?

Yeah I think it should be good. I just hadn't time to come back to this, because I'm on holiday right now.

**Comment by seven1m** [comment]
Rebased against master. Once the tests pass, I'll merge it.

**Comment by seven1m** [comment]
Ahhh, the error returned! (This is a good thing, maybe I'll figure it out now.)

**Pull request merged**